index.js

```diff
@@ -139,8 +139,18 @@ module.exports = function(version, _options) {
             }
             name = name.replace(' (' + step.ref + ')', '');
 
-            if (name && ref && name !== ref) {
+            // In attempt to avoid using the highway name of a way,
+            // check and see if the step has a class which should signal
+            // the ref should be used instead of the name.
+            var wayIsOnHighway = false;
+            if (options && options.classes) {
+                wayIsOnHighway = options.classes.some((className) => ['toll', 'highway'].indexOf(className) > -1);
```

Comment:
A toll road isn’t necessarily a nameless road. In fact, I’d argue that toll roads in the U.S. are more identifiable by their names than by their route numbers, even though OpenStreetMap tends to put acronyms in `ref` for map display purposes. Consider:

* “[New Jersey Turnpike](https://en.wikipedia.org/wiki/New_Jersey_Turnpike)”, not “NJTP”
* “[Garden State Parkway](https://en.wikipedia.org/wiki/Garden_State_Parkway)”, not “GSP”
* “[Dallas North Tollway](https://en.wikipedia.org/wiki/Dallas_North_Tollway)”, not “DNT”
* “[President George Bush Turnpike](https://en.wikipedia.org/wiki/President_George_Bush_Turnpike)” and “PGBT” are interchangeable
* “[Toll 23](https://en.wikipedia.org/wiki/Florida_State_Road_23)”, not “First Coast Expressway”
* “I 10 Toll”, not “[Katy Tollway](https://en.wikipedia.org/wiki/Katy_Tollway)”

For the most part, these toll roads are already tagged as `motorway`s. However, [Chickasaw Turnpike](https://en.wikipedia.org/wiki/Chickasaw_Turnpike) is tagged as a `trunk`; it would be better to call it “Chickasaw Turnpike” than “OK 7 Spur”.

More broadly, we shouldn’t give so much prominence to purely alphabetic refs, which may be mere abbreviations of the full name. For example, all the [New York Parkways](https://en.wikipedia.org/wiki/Parkways_in_New_York) have acronyms in `ref`: it should be “[Saw Mill River Parkway](https://en.wikipedia.org/wiki/Saw_Mill_River_Parkway)”, not “SMP”.

Given these examples, I think we should use the following logic, at least until we’re able to unambiguously identify routes based on route relations:

1. If the classes contain `motorway` (`highway`?) and the ref contains a digit, say just the ref.
2. If the classes contain `motorway` and the ref contains no digit, say only the name.
3. If the classes don’t contain `motorway`, say the name followed by the ref.

If you’re concerned that we’d be tossing out too many refs by requiring a digit, we could instead check whether the `ref` is a single, all-caps word – so `GSP`, not [`Orange`](https://en.wikipedia.org/wiki/Allegheny_County_belt_system#Orange_Belt).

I think these heuristics would work well for North America. Perhaps @freenerd could comment on whether they make any sense in Europe.